### PR TITLE
Deps installation

### PR DESCRIPTION
**Pull request by MeirShpilraien**

### CODE REVIEW

File: src/redisgears_python.c [hunk 1]
```diff
@@ -65,145 +65,367 @@ static void* RedisGearsPy_PyCallbackDeserialize(FlatExecutionPlan* fep, Gears_Bu
...
+            ++c;
+        }
+        filesInDir = array_append(filesInDir, ' ');
     }
-    array_free(requirementsList);
+    filesInDir[array_len(filesInDir) - 1] = '\0';
+
+    int exitCode = ExecCommand(NULL, "/bin/bash -c \"source %s/bin/activate;cd %s;python -m pip install %s\"", venvDir, req->basePath, filesInDir);
+    array_free(filesInDir);
+    if(exitCode != 0){
```
Comment (rafie) [inline src/redisgears_python.c]: return exitCode == 0;

File: src/redisgears_python.c [hunk 2]
```diff
@@ -65,145 +65,367 @@ static void* RedisGearsPy_PyCallbackDeserialize(FlatExecutionPlan* fep, Gears_Bu
...
-        PythonSessionCtx* session = Gears_dictFetchValue(SessionsDict, id);
-        if(session){
-            PythonSessionCtx_FreeRequirementsList(requirementsList);
-            return PythonSessionCtx_ShellowCopy(session);
+static void PythonRequirementCtx_Free(PythonRequirementCtx* reqCtx){
+    if(--reqCtx->refCount){
+        return;
+    }
+
+    ExecCommand(NULL, "rm -rf %s", reqCtx->basePath);
```
Comment (rafie) [inline src/redisgears_python.c]: reqCtx->basePath == "/" scares me.

File: src/redisgears_python.c [hunk 3]
```diff
@@ -65,145 +65,367 @@ static void* RedisGearsPy_PyCallbackDeserialize(FlatExecutionPlan* fep, Gears_Bu
...
+
+            if(!PythonRequirementCtx_InstallRequirement(req)){
+                PythonRequirementCtx_Free(req);
+                PythonSessionCtx_Free(session);
+                session = NULL;
+                break;
+            }
+        }
+
+        session->requirements = array_append(session->requirements , req);
```
Comment (rafie) [inline src/redisgears_python.c]: session can be NULL here.

Comment (MeirShpilraien) [reply src/redisgears_python.c]: How can it be NULL?

File: src/redisgears_python.c (hunk 2 above)
Comment (MeirShpilraien) [reply src/redisgears_python.c]: What do you suggest?

File: src/redisgears_python.c (hunk 3 above)
Comment (rafie) [reply src/redisgears_python.c]: You're right, it seemed to me it was outside the for.

File: src/redisgears_python.c (hunk 2 above)
Comment (rafie) [reply src/redisgears_python.c]: From looking at the construction of basePath, it looks like we should check that it doesn't match /+. Maybe do it in a function and call it in creation site and rm -rf site just to be sure we don't rack the system.

**Pull request merged**